### PR TITLE
Bug fix 3.4/reduce arangosearch rwlock waits

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

this is a quick mitigation only, which reduces maximum wait time from 1
second to 100 milliseconds without changing other behavior.

the main problem of notifying pending writers without successfully
acquiring the required mutex still needs proper addressing.

- [ ] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [x] Bug-Fix for a *released version* (did you remember to port this to all relevant release branches?)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)

### Testing & Verification

This change is a trivial rework / code cleanup without any test coverage.

https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/5527/

### CODE REVIEW

**Comment by jsteemann** [comment]
tests blue

**Comment by jsteemann** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/5540/

File: 3rdParty/iresearch/core/utils/async_utils.cpp [hunk 1]
```diff
@@ -179,21 +178,20 @@ void read_write_mutex::unlock(bool exclusive_only /*= false*/) {
...
     ADOPT_SCOPED_LOCK_NAMED(mutex_, lock);
-    static std::thread::id unowned;
 
     if (exclusive_only) {
-      ++concurrent_count_; // aquire the read-lock
+      ++concurrent_count_; // acquire the read-lock
     }
 
-    exclusive_owner_.store(unowned);
+    exclusive_owner_.store(std::thread::id());
```
Comment (mpoeter) [inline 3rdParty/iresearch/core/utils/async_utils.cpp]: This thread is currently unlocking the mutex, so why are we storing our thread-id as owner here (again)?

Comment (jsteemann) [reply 3rdParty/iresearch/core/utils/async_utils.cpp]: No, this is not storing the current thread's id.
This is the constructor of `std::thread::id()`, which, according to [cppreference](https://en.cppreference.com/w/cpp/thread/thread/id/id) 
> Default-constructs a new thread identifier. The identifier does not represent a thread.

Comment (mpoeter) [reply 3rdParty/iresearch/core/utils/async_utils.cpp]: 🤦‍♂ wer lesen kann ist klar im Vorteil... 🙄

**Pull request merged**